### PR TITLE
[8.0] port account payment blocking from 7.0

### PR DESCRIPTION
**Pull request by sbidoul**

### CODE REVIEW

File: account_payment_blocking/README.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License
+
+account_payment_blocking
+========================
+
+This module was written to extend the functionality of payment orders
+to block invoices under litigation to be presented for inclusion un payment orders.
```
Comment (lmignon) [inline account_payment_blocking/README.rst]: typo '**un** payment orders'

File: account_payment_blocking/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    'test': [
+    ],
+    'demo': [
+    ],
+    'js': [
+    ],
+    'qweb': [
+    ],
+    'css': [
+    ],
```
Comment (lmignon) [inline account_payment_blocking/__openerp__.py]: Do we need to keep empty entries?

File: account_payment_blocking/tests/__init__.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from . import test_account_banking_payment_blocking
+
+checks = [
```
Comment (lmignon) [inline account_payment_blocking/tests/__init__.py]: in V8 `checks =` is no more requried

File: account_payment_blocking/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+#
+##############################################################################
+{
+    'name': 'account banking payment blocking',
+    'version': '0.1',
+    'category': 'Banking addons',
+    'summary': """
+        Prevent invoices under litigation to be proposed in payment orders.
+    """,
+    'author': 'ACSONE SA/NV',
```
Comment (lmignon) [inline account_payment_blocking/__openerp__.py]: OCA misses in the list of authors

**Comment by sbidoul** [comment]
@lmignon thanks for the review. I adapted.

**Comment by lmignon** [comment]
:+1:  (code review) Thank you!

File: account_payment_blocking/tests/test_account_banking_payment_blocking.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import openerp.tests.common as common
+from openerp import netsvc
```
Comment (adrienpeiffer) [inline account_payment_blocking/tests/test_account_banking_payment_blocking.py]: Can you directly use workflow ?

**Comment by adrienpeiffer** [comment]
:+1: (Code review). Thanks !

File: account_payment_blocking/README.rst [hunk 6]
```diff
@@ -0,0 +1,65 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License
+
+account_payment_blocking
```
Comment (pedrobaeza) [inline account_payment_blocking/README.rst]: Put a proper title.

**Comment by pedrobaeza** [comment]
You can use new OCA's generic module icon: https://github.com/OCA/maintainer-tools/blob/master/template/module/static/description/icon.png

**Comment by pedrobaeza** [comment]
Two little style remarks.

File: account_payment_blocking/README.rst [hunk 7]
```diff
@@ -0,0 +1,65 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License
```
Comment (yvaucher) [inline account_payment_blocking/README.rst]: Please change this line to

```
    :alt: License: AGPL-3
```

File: account_payment_blocking/__init__.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Account Payment Blocking module for OpenERP
```
Comment (yvaucher) [inline account_payment_blocking/__init__.py]: for Odoo ?

File: account_payment_blocking/__openerp__.py [hunk 9]
```diff
@@ -0,0 +1,43 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Account Payment Blocking module for OpenERP
```
Comment (yvaucher) [inline account_payment_blocking/__openerp__.py]: for Odoo ?

File: account_payment_blocking/model/account_invoice.py [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    _inherit = "account.invoice"
+
+    def _get_move_line(self, cr, uid, invoice_id, context=None):
+        return self.pool.get('account.move.line')\
+            .search(cr, uid, [('account_id.type', 'in',
+                              ['payable', 'receivable']),
+                              ('invoice.id', '=', invoice_id)])
+
+    def _set_move_blocked(self, cr, uid, ids, name, field_value, arg,
+                          context=None):
```
Comment (yvaucher) [inline account_payment_blocking/model/account_invoice.py:37]: Please ensure

```
if context is None:
     context = {}
```

File: account_payment_blocking/__openerp__.py [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+#    Copyright (C) 2014 ACSONE SA/NV (http://acsone.eu)
```
Comment (yvaucher) [inline account_payment_blocking/__openerp__.py]: 2014-2015 ?

File: account_payment_blocking/model/account_invoice.py [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+
+
+class account_invoice(orm.Model):
+    _inherit = "account.invoice"
+
+    def _get_move_line(self, cr, uid, invoice_id, context=None):
+        return self.pool.get('account.move.line')\
+            .search(cr, uid, [('account_id.type', 'in',
+                              ['payable', 'receivable']),
+                              ('invoice.id', '=', invoice_id)])
```
Comment (yvaucher) [inline account_payment_blocking/model/account_invoice.py]: missing context on search

**Comment by yvaucher** [comment]
Just few remarks + @pedrobaeza remarks otherwise :+1:

**Comment by sbidoul** [comment]
@yvaucher @pedrobaeza thanks for the review. I took care of your comments.

**Comment by yvaucher** [comment]
Thanks :+1:

**Comment by lmignon** [comment]
:+1:

**Comment by pedrobaeza** [comment]
:+1:

**Pull request merged**